### DISCUSSION TITLE
How to check if BMP is a monochrome or not ?

### DISCUSSION THREAD

**Discussion by piyushmeshram2014** [mono/SkiaSharp#2179] [Q&A]
Hi Team, 
I am new to SkiaSharp and looking for a way to validate if my bmp image is monochrome or not?
I am doing it something like this
```
var bmpImage = SKBitmap.Decode(contentData);
if(bmpImage.Info.BitsPerPixel != 1)
   throw new Exception(..)
```
The BitPerPixel always gives "32" , even for a monochrome image where I am expecting it to be "1".Kindly help

**Comment by mattleibow** [comment]
You can try use `SKCodec` directly. Using `SKBitmap` actually decodes and consumes memory. And then it decodes into a ready to draw format - hence the 32 bits.

The code will allow you to open the image and read info about the image without actually consuming memory or changing bits.

**Reply by piyushmeshram2014** [reply]
It still gives 32 with SKCodec

**Comment by IldarKhayrutdinov** [comment]
Maybe help you `ImageSharp` .

[BmpMetadata](https://github.com/SixLabors/ImageSharp/blob/13897ae4c59da50c37a017a4d21fbe80ae1bdde1/src/ImageSharp/Formats/Bmp/BmpMetadata.cs#L36) contains accurate [BmpBitsPerPixel](https://github.com/SixLabors/ImageSharp/blob/13897ae4c59da50c37a017a4d21fbe80ae1bdde1/src/ImageSharp/Formats/Bmp/BmpBitsPerPixel.cs).


Reading only image info example (without fully loading):
`IImageInfo imageInfo = Image.Identify(stream);`
https://github.com/SixLabors/ImageSharp/blob/13897ae4c59da50c37a017a4d21fbe80ae1bdde1/tests/ImageSharp.Tests/Formats/Bmp/BmpMetadataTests.cs#L43-L45
